sdk/src/hard_forks.rs

```diff
@@ -22,7 +22,7 @@ impl HardForks {
         } else {
             self.hard_forks.push((new_slot, 1));
         }
-        self.hard_forks.sort();
+        self.hard_forks.sort_unstable();
```

Comment:
Can you remove all the `sort_unstable()` changes please.  Unstable sorts can break consensus, so we'll need to carefully consider each sort -> sort_unstable change individually

Reply:
All replaces of `sort` to `sort_unstable` in this PR targets exactly `Vec` of `Slot`. `Slot` is `Copy` type. And for `Vec<T: Copy>` there is no difference between `sort` and `sort_unstable`.

Reply:
I can revert all `sort_unstable()` back to `sort()` and marks them as `#[allow(clippy::stable_sort_primitive)]`.

Reply:
My preference is to do that.   I'm worried also about a future refactoring that might change result in type change where `sort_unstable()` is no longer equivalent to `sort()`, and accidentally breaking consensus.  `clippy::stable_sort_primitive` is borderline harmful IMO, the implementation of `sort()` should just be improved for the types were `sort_unstable()` and `sort()` are equivalent.

Reply:
Agree. Fixed.